fvm/evm/precompiles/arch_test.go

```diff
@@ -37,6 +38,34 @@ func TestArchContract(t *testing.T) {
 		require.Error(t, err)
 	})
 
+	t.Run("test get random source", func(t *testing.T) {
+		address := testutils.RandomAddress(t)
+		rand := uint64(1337)
+		pc := precompiles.ArchContract(
+			address,
+			nil,
+			nil,
+			func(u uint64) (uint64, error) {
+				return rand, nil
```

Comment:
I might add a line here to check if the input that is received is what is expected.

Reply:
good idea yeah